src/js/game/LevelMVC/LevelModel.js

```diff
@@ -583,21 +592,37 @@ module.exports = class LevelModel {
     return entity.canPlaceBlock(blockAtPosition);
   }
 
+  canPlaceBlockDirection(blockType = "", entity , direction) {
+    if (entity.isOnBlock) {
+      return false;
+    }
+    let plane = this.getPlaneToPlaceOn(this.getMoveDirectionPosition(entity, direction), entity);
+    if (plane === this.groundPlane) {
+      if (blockType === "redstoneWire" || blockType.substring(0,5) === "rails" && this.groundPlane.getBlockAt(this.getMoveDirectionPosition(entity, direction))) {
+        return false;
+      }
+    }
+    return this.getPlaneToPlaceOn(this.getMoveDirectionPosition(entity, direction), entity) !== null;
+  }
+
   canPlaceBlockForward(blockType = "", entity = this.player) {
```

Comment:
Same feedback as above — I think this becomes a special case of `canPlaceBlockDirection` (i.e. this method should just call that implementation).